### PR TITLE
Add custom data allocation support.

### PR DESCRIPTION
**Pull request by anthonyvercolano**
Add support for custom allocation.

### CODE REVIEW

File: provisioning/device/src/tpm_registration.ts [hunk 1]
```diff
@@ -26,6 +28,7 @@ export class TpmRegistration extends EventEmitter implements RegistrationClient
     this._idScope = idScope;
     this._pollingStateMachine = new PollingStateMachine(transport);
 
+
```
Comment (pierreca) [inline provisioning/device/src/tpm_registration.ts]: nit: unnecessary whitespace

File: provisioning/device/src/tpm_registration.ts [hunk 2]
```diff
@@ -224,6 +227,15 @@ export class TpmRegistration extends EventEmitter implements RegistrationClient
 
   }
 
+  /**
+   * Sets the custom payload for registration.
```
Comment (pierreca) [inline provisioning/device/src/tpm_registration.ts]: >for registration. [](start = 29, length = 17)

"that will be sent to the custom allocation policy implemented in an Azure Function (for more info see <link to docs>).

File: provisioning/transport/amqp/devdoc/amqp_requirements.md [hunk 3]
```diff
@@ -26,6 +26,12 @@ These methods are used by the other objects of the SDK but are not public API fo
 and shall set the password to the passed in sas token.
  **]**
 
+**SRS_NODE_PROVISIONING_AMQP_06_003: [** The `registrationRequest` will send a body in the message which contains a stringified JSON object with a `registrationId` property. **]**
+
+**SRS_NODE_PROVISIONING_AMQP_06_004: [** The `registrationRequest` will, if utilizing TPM attestation, send a `tpm` property with the endorsement and storage key in the JSON body. **]**
+
+**SRS_NODE_PROVISIONING_AMQP_06_005: [** The `registrationRequest` will, if utilizing custom allocate data, send a `payload` property in the JSON body. **]**
```
Comment (pierreca) [inline provisioning/transport/amqp/devdoc/amqp_requirements.md]: >allocate [](start = 93, length = 8)

allocation

File: provisioning/device/src/tpm_registration.ts (hunk 1 above)
Comment (anthonyvercolano) [reply provisioning/device/src/tpm_registration.ts]: Done.

---
In reply to: [272699366](https://github.com/Azure/azure-iot-sdk-node/pull/524#discussion_r272699366) [](ancestors = 272699366)

File: provisioning/device/src/constants.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ export class ProvisioningDeviceConstants {
   /**
    * apiVersion to use while communicating with service.
    */
-  static apiVersion: string = '2018-11-01';
+  static apiVersion: string = '2019-01-15';
```
Comment (BertKleewein) [inline provisioning/device/src/constants.ts]: > static apiVersion: string = '2019-01-15'; [](start = 1, length = 42)

has this API been deployed to all regions?  if not, this  change could break users.
